snactor/loader/group.py

```diff
@@ -0,0 +1,97 @@
+import os
+
+from snactor.loader import create_actor, load
+from snactor.registry import get_registered_actors, must_get_actor
+from snactor.utils import get_chan
+
+
+class UnresolvedDependenciesError(Exception):
+    """
+    UnresolvedDependenciesError is thrown when the dependencies between actors can't be automatically resolved
+    """
+    pass
+
+
+def _order_by_channels(channels, actors):
+    result = []
+    while actors:
+        scheduled = []
+        for idx, actor in enumerate(actors):
+            for channel in actor[0].inputs:
+                if channels[channel['name']]['producers']:
+                    break
+            else:
+                # for channel in actor[0].inputs:
+                #    channels[channel['name']]['consumers'].remove(actor[0].name)
+
+                for channel in actor[0].outputs:
+                    channels[channel['name']]['producers'].remove(actor[0].name)
+
+                scheduled.append(idx)
+                result.append(actor[0].name)
+
+        if not scheduled:
+            raise UnresolvedDependenciesError(
+                "Could not solve dependency order for '{}'".format(', '.join([a[0].name for a in actors])))
+
+        for idx in reversed(scheduled):
+            actors.pop(idx)
+
+    return result
+
+
+def load_as_group(name, location, tags=()):
+    """
+    load_as_group creates a group actor with the given `name`, with actors loaded from `location` and filtered by `tags`
```

Comment:
group.py:45:120: E501 line too long (120 > 119 characters). Failing on Travis because of this